MantidPlot/src/Mantid/MantidUI.cpp

```diff
@@ -459,10 +459,11 @@ void MantidUI::saveNexusWorkspace() { executeSaveNexus(); }
 @param workspaceName :: Name of the workspace to delete
 */
 void MantidUI::deleteWorkspace(const QString &workspaceName) {
-  auto alg = createAlgorithm("DeleteWorkspace");
-  alg->setLogging(false);
-  alg->setPropertyValue("Workspace", workspaceName.toStdString());
-  executeAlgorithmAsync(alg);
+  auto workspaceNameStdString = workspaceName.toStdString();
+  auto &ads = Mantid::API::AnalysisDataService::Instance();
+  if (ads.doesExist(workspaceNameStdString)) {
```

Comment:
Is it necessary to check the existence first? `DataService::remove` does nothing if the name does not exist so this results in a double look up most of the time.